superset/connectors/sqla/models.py

```diff
@@ -650,55 +651,52 @@ def geo_features_gen(values):
                 flt["val"] = new_flt_geo_values
                 flt["col"] = "geo"
                 col = "geo"
-                op = "geo_within"
-            col_obj = cols.get(col)
+                col_obj = cols.get(col)
+                features = flt.get('val')["features"]
+
+                shapes = [geometry.shape(feature["geometry"]) for feature in features]
+                total_shape = ops.cascaded_union(shapes)
+                eq = total_shape
+                where_clause_and.append(
+                    sa.func.ST_COVERS(
+                        sa.func.ST_GeogFromText(eq.wkt),
+                        sa.cast(cols.get("geo").get_sqla_col(), Geometry)
+                    )
+                )
             if col_obj:
                 is_list_target = op in ('in', 'not in')
-                if op in ["geo_within"]:
-                    features = flt.get('val')["features"]
-
-                    shapes = [geometry.shape(feature["geometry"]) for feature in features]
-                    total_shape = ops.cascaded_union(shapes)
-                    eq = total_shape
-                    where_clause_and.append(
-                        sa.func.ST_COVERS(
-                            sa.func.ST_GeogFromText(eq.wkt),
-                            sa.cast(cols.get("geo").get_sqla_col(), Geometry)
-                        )
-                    )
+                eq = self.filter_values_handler(
```

Comment:
I think that breaking this flow and extracting all of `geo_within` login above breaks filtering. I have tested it locally and population data is not updated properly after picking raion or oblast. It works fine at master.
I have already tried to do a similar thing as you did but in the end needed to revert to split the `geo_within` logic to before and after `if col_obj:` line.

Reply:
please test  with a geofilterable checkbox checked